pkg/azure/client.go

```diff
@@ -116,41 +118,86 @@ func (az *azClient) SetAuthorizer(authorizer autorest.Authorizer) {
 	az.routeTablesClient.Authorizer = authorizer
 	az.groupsClient.Authorizer = authorizer
 	az.deploymentsClient.Authorizer = authorizer
+	az.roleAssignmentsClient.Authorizer = authorizer
 }
 
-func (az *azClient) GetGateway() (response n.ApplicationGateway, err error) {
-	err = utils.Retry(retryCount, retryPause,
+func (az *azClient) WaitForAccess(resourceID string, role RoleDefinition) {
+	utils.Retry(-1, retryPause,
```

Comment:
Is this a endless wait ? Shall we atleast somewhere print the iteration count

Reply:
Would it be useful as this is an endless wait ?